UPGRADING.md

```diff
@@ -0,0 +1,56 @@
+UPGRADING
+--------
+
+This file documents how to upgrade an existing BHIMA installation to a new release.  To learn about the releases process, see [RELEASES](./RELEASES.md).
+
+Upgrading an existing installation takes three steps:
+
+1. Preparing the BHIMA software
+2. Preparing the database
+3. Running the upgrade
+
+Steps 1 and 2 can happen in parallel and can be done at a leasurely pace.  The third step is destructive and will commit the changes, deploying the new version at the location.
+
+These steps describe the typical deployment setup.  We assume that the BHIMA software is deployed in `$HOME/apps/`, `$VERSION` will be the next version name, and will refer to `$DATABASE` as the database name.
+
+## Step 1: Peparing the BHIMA software
+
+1. Go to https://github.com/ima-worldhealth/bhima/releases/latest in your web browser.
+2. Copy the link to the `$VERSION.tar.gz` file.
+2. Connect to the server with SSH
+3. Start a `screen` session in case the connection drops during the upgrade.
+4. Change directory into `$HOME/apps/`.
+5. Download the latest release with `wget -c "$URL"`.  This will create `$VERSION.tar.gz` in the `$HOME/apps/` directory.
+6. Unzip the release file with `tar xf $VERSION.tar.gz`.  This will create the folder `bhima-$RELEASE/` in the `$HOME/apps/` directory.
+7. Copy the `.env` file from the current application into the `bhima-$RELEASE/bin/` folder and make any changes necessary to `.env`.
+8. Copy the contents (but not sub-directories) from the `bhima-$RELEASE/bin/` folder into the `bhima-$RELEASE/` folder.  This can be done with `cp $HOME/apps/bhima-$RELEASE/bin/* $HOME/apps/bhima-$RELEASE/bin/`
+9. Change directory into the release directory with `cd $HOME/apps/bhima-$RELEASE/`
+10. Install the `node_modules/` in the `bhima-$RELEASE/` folder with the command `NODE_ENV=production yarn`.
+
+The BHIMA upgrade is not prepared.
+
+## Step 2: Preparing the database
+
+Locally:
+1. Download the latest production database from the BHIMA backups server for the site you are upgrading.
+2. Build it locally.
+3. Change the environmental variable `$DB_NAME` in the `.env` file in the bhima repository to the name of the site you are upgrading.
+4. Run `yarn migrate` to create a migration script `migration-$DATABASE.sql`.
+5. Copy in the migration files from up to the present version from the `server/models/migrations/` folder(s).  You can use `cat` for this.
+6. Test the database migration script `mysql $DATABASE < migration-$DATABASE.sql`
+7. If it works, copy the database over to the remote server with `scp`.  `scp migration-$DATABASE.sql bhima@target.bhi.ma:~/`.
+
+The database is now prepared.
+
```

Comment:
Should you check to see if any clients are connected to the server?    Or maybe the point here is that you should coordinate with the staff at the production site unless you know they will all be gone (eg, late night upgrades).

Reply:
I'm hoping this is already done.  I'm not sure we need to cover that in these steps, since they generally happen at different times (you plan with staff way, way in advance of touching this document).